crates/chia-sdk-driver/src/layers/did_layer.rs

```diff
@@ -3,170 +3,130 @@ use chia_puzzles::{
     did::{DidArgs, DidSolution, DID_INNER_PUZZLE_HASH},
     singleton::SingletonStruct,
 };
-use clvm_traits::ToNodePtr;
 use clvm_traits::{FromClvm, ToClvm};
 use clvm_utils::{CurriedProgram, ToTreeHash, TreeHash};
 use clvmr::{Allocator, NodePtr};
 
-use crate::{DriverError, Puzzle, PuzzleLayer, SpendContext};
+use crate::{DriverError, Layer, Puzzle, SpendContext};
 
 #[derive(Debug)]
-pub struct DidLayer<M, IP> {
-    pub launcher_id: Bytes32,
-    pub recovery_did_list_hash: Bytes32,
+pub struct DidLayer<M, I> {
+    pub singleton_struct: SingletonStruct,
+    pub recovery_list_hash: Bytes32,
```

Comment:
should we make this `Option<Bytes32>`? if there isn't a recovery list, a `()` default would save 31 bytes from the puzzle reveal

Reply:
I'd have to update chia_rs to do that, but I can make a PR for that (and actually it'd technically save 32 since `()` is zero length)

Reply:
https://github.com/Chia-Network/chia_rs/pull/667